server/src/utilities/generateComponent/helpers/generateReact.ts

```diff
@@ -0,0 +1,62 @@
+import { Node, SyntaxKind, SourceFile } from 'ts-morph';
+
+import MezzuriteComponent from '../../../models/MezzuriteComponent';
+
+function generateReact (filePath: string, sourceFile: SourceFile): MezzuriteComponent {
+  let component = null;
+
+  if (filePath != null && sourceFile != null) {
+    const exportContents = getExportContents(sourceFile);
+
+    if (exportContents != null) {
+      let hasWithMezzurite = false;
+      let name = null;
+
+      if (exportContents.getKind() === SyntaxKind.Identifier) {
+        name = exportContents.getText();
+      } else {
+        const identifiers = exportContents.getDescendantsOfKind(SyntaxKind.Identifier);
+        const lastIdentifier = identifiers[identifiers.length - 1];
+
+        if (lastIdentifier != null) {
+          name = lastIdentifier.getText();
+        }
+
+        hasWithMezzurite = exportContents.getText().indexOf('withMezzurite') > -1;
+      }
+
+      component = {
+        checks: {
+          hasWithMezzurite
+        },
+        filePath,
+        name,
+        type: 'react'
+      };
+    }
+  }
+
+  return component;
+}
+
+function getExportContents (sourceFile: SourceFile): Node {
+  let exportContents = null;
+  const fileContents = sourceFile.getChildSyntaxList();
```

Comment:
Is it okay to not check if `sourceFile` is `null` or `undefined`?

Reply:
That's a good call out. These two function should not be bound together. Resolved via https://github.com/Microsoft/vscode-mezzurite/pull/32/commits/fbbfa606958f986bc5a595e6dca07585ead5c401.